### ISSUE TITLE
Battles can fail to end when a team has zero reinforcements

### ISSUE THREAD

**Issue by AndrewTheel** [DarklightGames/DarkestHour#879]
Yikes.

**Comment by AndrewTheel** [comment]
I think I've fixed this, but also put in a fail safe in case.

**Comment by AndrewTheel** [comment]
Okay I cleaned up the code and removed the fail safe, it should fix the bug. I searched and didn't see any thing that edited reinforcements outside the `ModifyReinforcements` during the round. As long as that is in fact the case, the bug will be fixed.

**About the bug:**
The bug is that the check after it subtracts a reinforcement was failing because it though there was still reinforcements remaining, because it did before the reinforcement edit. Once that check was skipped another system prevented additional players from trying to spawn and thus "this check" in modify reinforcements wasn't being run anymore. The fix is to fix the logic bug in the `ModifyReinforcements` function, however there is always a possibility that something else removes reinforcements without the `ModifyReinforcements` function. This is why I put in the "fail safe" in timer which checks if reinforcements are out and then properly ends the round. However, the "fail safe" is just redundant and we want less timer logic, so I removed. If the bug still exists its because something else is changing reinforcements outside the `ModifyReinforcements` function and during round live. At which point we will need to bitch slap.
